extension/scripts/Selector/SelectorTable.js

```diff
@@ -106,42 +119,52 @@ var SelectorTable = {
         return result;
     },
 
+    isEmptyObject: function (obj) {
+    for (var i in obj) {
+        if (obj.hasOwnProperty(i)) {
+            return false;
+        }
+    }
+    return true;
+    },
     _getData: function (parentElement) {
 
         var dfd = $.Deferred();
 
         var tables = this.getDataElements(parentElement);
-
+        var verticalTable = this.verticalTable;
+        if (this.verticalTable === undefined) {
+            verticalTable = $("#edit-selector [name=verticalTable]").is(":checked");
```

Comment:
No need to extract this again here. It should be already in selector object. If it is not then there is a bug in controller.